src/SmartTransactionsController.ts

```diff
@@ -330,42 +396,59 @@ export default class SmartTransactionsController extends BaseController<
     });
   }
 
-  async updateSmartTransactions() {
+  async updateSmartTransactions({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  } = {}): Promise<void> {
     const { smartTransactions } = this.state.smartTransactionsState;
-    const { chainId } = this.config;
-
-    const currentSmartTransactions = smartTransactions?.[chainId];
+    const chainId = this.getChainId({ networkClientId });
+    const smartTransactionsForChainId = smartTransactions?.[chainId];
 
-    const transactionsToUpdate: string[] = currentSmartTransactions
+    const transactionsToUpdate: string[] = smartTransactionsForChainId
       .filter(isSmartTransactionPending)
       .map((smartTransaction) => smartTransaction.uuid);
 
     if (transactionsToUpdate.length > 0) {
-      this.fetchSmartTransactionsStatus(transactionsToUpdate);
+      this.fetchSmartTransactionsStatus(transactionsToUpdate, {
+        networkClientId,
+      });
     }
   }
 
-  async confirmSmartTransaction(smartTransaction: SmartTransaction) {
+  async #confirmSmartTransaction(
+    smartTransaction: SmartTransaction,
+    {
+      chainId,
+      ethQuery,
+    }: {
+      chainId: Hex;
+      ethQuery: EthQuery;
+    },
+  ) {
     const txHash = smartTransaction.statusMetadata?.minedHash;
     try {
-      const transactionReceipt =
-        await this.ethersProvider.getTransactionReceipt(txHash);
-      const transaction = await this.ethersProvider.getTransaction(txHash);
-      const maxFeePerGas = transaction.maxFeePerGas?.toHexString();
-      const maxPriorityFeePerGas =
-        transaction.maxPriorityFeePerGas?.toHexString();
+      const transactionReceipt: {
+        maxFeePerGas?: string;
+        maxPriorityFeePerGas?: string;
+        blockNumber: string;
+      } | null = await query(ethQuery, 'getTransactionReceipt', [txHash]);
+
+      const transaction: {
+        maxFeePerGas?: string;
+        maxPriorityFeePerGas?: string;
+      } = await query(ethQuery, 'getTransaction', [txHash]);
```

Comment:
Looks like `getTransaction` was another Ethers convenience method. Should this be `getTransactionByHash`? If so then Infura says [that this can also be `null`](https://docs.infura.io/networks/ethereum/json-rpc-methods/eth_gettransactionbyhash):

``` suggestion
      } | null = await query(ethQuery, 'getTransactionByHash', [txHash]);

Reply:
👍 fixed here: https://github.com/MetaMask/smart-transactions-controller/pull/210/commits/9c4a772cc5dbff42e11bdad85d1076b94f751c9d